### PR TITLE
LPS-191628 Simplify the logic and sort the expectedToRecipients, notificationQueueEntries and emailAddresses arrays to be validated in the correct order

### PR DESCRIPTION
**Pull request by seltong**
Follow-up: https://github.com/brianchandotcom/liferay-portal/pull/139280
Related issue: [LPS-191628](https://liferay.atlassian.net/browse/LPS-191628)

### CODE REVIEW

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 1]
```diff
@@ -120,19 +121,36 @@ public void testSendNotification() throws Exception {
 
 		_addObjectEntry();
 
-		notificationQueueEntries =
+		List<String> expectedToRecipients = ListUtil.sort(
+			Arrays.asList(user1.getEmailAddress(), user2.getEmailAddress()));
```
Comment (murilostodolni) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:125]: Can we move to before first use in line 149?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 2]
```diff
@@ -241,34 +259,16 @@ private void _assetNotificationQueueEntry(
...
 
-					break;
-				}
-			}
-
-			Assert.assertTrue(
-				actualToRecipient + " does not contain " + expectedToRecipient,
-				containsEmailAddress);
-		}
+		Assert.assertArrayEquals(expectedEmailAddresses, actualEmailAddresses);
```
Comment (murilostodolni) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:271]: I don't know if the best is:
1. Split twice and after Sort twice OR
2. Split and Sort in sequence. Something like this:

 ```java
String[] actualEmailAddresses = StringUtil.split(
	String.valueOf(notificationRecipientSettingsMap.get("to")));

Arrays.sort(actualEmailAddresses);

String[] expectedEmailAddresses = StringUtil.split(expectedToRecipient);

Arrays.sort(expectedEmailAddresses);

Assert.assertArrayEquals(expectedEmailAddresses, actualEmailAddresses);
```

I see the two approach in the portal :/

**Review by murilostodolni — CHANGES_REQUESTED** [review]
Hey @seltong 

I finished my review. I have only 2 questions. Let me know if you have questions.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 2 above)
Comment (murilostodolni) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:271]: Yes, I agree that second approach seems easier to read! Thank you

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

**Review by murilostodolni — APPROVED** [review]
LGTM!

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Pull request closed without merging**